### ISSUE TITLE
Invited Members: Bulk Invite Users 

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#17453]
📖 **As an Admin, I want my experience with invitations to be as pleasing and straightforward as possible, so that I feel motivated to use them.**

✅ Tasks to complete:
- [ ]  Alter the new invite member flow to bulk invite users ( new page workflow + bulk invite + account for validation + success and error states )

<img width="653" alt="Screenshot 2022-05-03 at 20 28 19" src="https://user-images.githubusercontent.com/2786819/166518779-7578804b-db12-4eb4-84be-5136e25f5e71.png">
<img width="631" alt="Screenshot 2022-05-03 at 20 29 07" src="https://user-images.githubusercontent.com/2786819/166518918-e275c22b-0ca7-4d40-b783-d32f0ec0bd06.png">
<img width="856" alt="Screenshot 2022-05-03 at 20 29 31" src="https://user-images.githubusercontent.com/2786819/166518980-d10e66df-0ca1-4d96-b966-04e47b185e70.png">

**Comment by Ridhwana** [comment]
@aitchiss do you have any initial thoughts or tips for the Preact Component for the email addresses that I should take into consideration? It would probably need to be custom built right (i.e. without a library)? I see it being similar in UI as the MultipleSelectAutoComplete (in the way that the UI looks and without the autocompletion).

**Comment by aitchiss** [comment]
Yeah I've been mulling this over @Ridhwana.. one option I was weighing up is altering the `MultiSelectAutocomplete` so that it supports both autocomplete and non-autocomplete behaviours 🤔  Because, like you say the behaviours and UI we're looking for are _so_ similar.

For example, we could rename the component from `MultiSelectAutocomplete` to something like `MultiSelectInput` and give it a `autocomplete` boolean prop to determine whether it's a 'normal' input or a 'suggesting' input. The thing that gives me pause though is that for this particular use case we need to also add in validation _after_ the user makes their selection, including error UI etc. And the `MultiSelectAutocomplete` component is already quite complicated.

TL;DR - for now perhaps the best move is to make a separate Preact component (e.g. `MultiEntryInput` or some other name), and keep an eye for any abstractions we can make in the `MultiSelectAutocomplete` to support that? What do you think?

I'd probably recommend that we break this task down further into:

- Create a crayons component in Storybook
- Implement it in the admin area (which will need to include some kind of progressive enhancement since it's Preact)

**Comment by Ridhwana** [comment]
> TL;DR - for now perhaps the best move is to make a separate Preact component (e.g. MultiEntryInput or some other name), and keep an eye for any abstractions we can make in the MultiSelectAutocomplete to support that? What do you think? We can definitely borrow a lot of styles, keyboard behaviours etc from the MultiSelectAutocomplete.

> I'd probably recommend that we break this task down further into:
> 
> Create a crayons component in Storybook
> Implement it in the admin area (which will need to include some kind of progressive enhancement since it's Preact)

I like this and think it's the most logical way forward ✨. Thank you so much for jotting down your thoughts! I'll start off by taking a stab at the crayons component which means I may ping you from time to time for some support if that's okay?

**Comment by aitchiss** [comment]
Sounds good! In the end, if you see they're more alike then we can always later refactor to combine them 🙂
